src/Tempest/Container/src/GenericContainer.php

```diff
@@ -105,6 +106,9 @@ public function singleton(string $className, mixed $definition, ?string $tag = n
 
     public function config(object $config): self
     {
+        if ($config instanceof DatabaseConfig) {
+            $this->singleton(DatabaseConfig::class, $config);
+        }
```

Comment:
Unsure, will think of a more generalized approach

Reply:
The approach I was thinking about was something like this (dummy code):

```php
    public function config(object $config): self
    {
        $this->singleton($config::class, $config);
        
        foreach (reflect($config)->getInterfaces() as $interface)
        {
            $this->singleton($interface, $config);
        }

        return $this;
    }
```

Reply:
Thanks, was thinking about something like this as well. Nice